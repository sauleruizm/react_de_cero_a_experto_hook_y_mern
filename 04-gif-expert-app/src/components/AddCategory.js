import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ( e ) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');            
        }
        //console.log('submit echo');
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Add category</h2>
            <p>{inputValue}</p>
            <input type="text" 
                value={inputValue} 
                onChange= { handleInputChange }/ >
        </form>
    )
    
}

AddCategory.propTypes =  {
    setCategories: PropTypes.func.isRequired
}