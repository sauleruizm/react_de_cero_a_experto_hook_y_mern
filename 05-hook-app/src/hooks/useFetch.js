import { useEffect, useState } from 'react'

export function useFetch  (url) {


	const [state, setState] = useState({
		data: null,
		isLoading: true,
		hasError: null,
	})

	const getFetch = async () => {
		setState({
			...state,
			isLoading: true,
		})

		const resp = await fetch(url)
		const data = await resp.json()

		console.log('useFetch =>',data)
		setState({
			data: data,
			isLoading: false,
			hasError: null,
		})
	}

	useEffect(() => {
		getFetch()
	}, [url])

	//console.log('usefetch: return state=>',state)
	return {
		...state
	}
}

// export default useFetch