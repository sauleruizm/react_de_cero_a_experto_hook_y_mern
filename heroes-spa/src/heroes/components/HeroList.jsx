import { HeroCard } from './HeroCard'

import { getHeroesByPublisher } from '../helpers'
import { useMemo } from 'react'

export const HeroList = ({ publisher }) => {
	const heroes = useMemo( ()=> getHeroesByPublisher(publisher), [ publisher ] )

	return (
		<div className='row rows-cols-1 row-cols-md-3 g-3'>

			{heroes.map((hero) => {
				// console.table(hero)
				return <HeroCard key={hero.id}  {...hero} />
			})}
		</div>
	)
}
