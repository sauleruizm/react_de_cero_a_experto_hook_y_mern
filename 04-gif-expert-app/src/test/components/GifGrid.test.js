import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('pruebas en GifGrid', () => {
    
    const category = 'one Punch';

    test('debe mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();

    })
  
    test('debe de mostrar items cuandose carga imÃ¡genes useFetchGifs', () => {
    
        const gifs = [
            {id:'abc', url:'https:/test.img', title:'cualquier cosa'}
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
})
