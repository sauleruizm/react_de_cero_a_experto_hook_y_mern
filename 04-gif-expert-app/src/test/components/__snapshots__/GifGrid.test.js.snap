// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid debe de mostrar items cuandose carga im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https:/test.img"
    />
  </div>
</Fragment>
`;

exports[`pruebas en GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
